src/Parser/Rule/Common.idr

```diff
@@ -16,20 +16,20 @@ EmptyRule token ty = Grammar token False ty
 export
 location : {token : _} -> EmptyRule token (Int, Int)
 location
-    = do tok <- bounds peek
-         pure (tok.startLine, tok.startCol)
+    = do tok <- removeIrrelevance <$> bounds peek
+         pure $ start tok
 
 export
 endLocation : {token : _} -> EmptyRule token (Int, Int)
 endLocation
-    = do tok <- bounds peek
-         pure (tok.endLine, tok.endCol)
+    = do tok <- removeIrrelevance <$> bounds peek
+         pure $ end tok
```

Comment:
I still don't understand the point of this `removeIrrelevance <$>`:
the value returned by the parser is `end tok` and it does not contain
this irrelevance flag as it's just a pair of two `Int`s.

Reply:
https://github.com/idris-lang/Idris2/blob/a7cb2745bcc7cf6b920ff00eecb18372835c4a06/src/Text/Parser/Core.idr#L261
The link points to how values returned by two parsers are composed. Here we are actually composing `bounds peek` and `pure $ end tok`; the former carries bounds whereas the latter doesn't. In this case, bounds from `bounds peek` will be carried over to the ultimate value returned by the parser. 

By adding `removeIrrelevance` I am actually trying to void the bounds carried by `bounds peek` so that the returned value of the composed parser doesn't carry any bounds. I hope this makes sense.